### PR TITLE
add swift data

### PR DESCRIPTION
**Pull request by MaksimK193**

### CODE REVIEW

File: SwiftUIStudy/Core/Managers/SwiftDataManager.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    }
+
+    func delete<T>(_ model: T) {
+        guard let _model = model as? any PersistentModel else { return }
+        context.delete(_model)
+    }
+    
+    func save() {
+        try? context.save()
```
Comment (bguchinov) [inline SwiftUIStudy/Core/Managers/SwiftDataManager.swift:52]: I guess the better approach use the same syntax, when you handling try catch things

File: SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    
+    var body: some View {
+        VStack {
+            HStack {
+                TextField("id", text: $id)
+                    .textFieldStyle(.roundedBorder)
+                TextField("name", text: $departmentName)
+                    .textFieldStyle(.roundedBorder)
+                Button(action: {
+                    let dep1 = Department(id: Int(id) ?? 1, name: departmentName, employees: [])
```
Comment (bguchinov) [inline SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift:23]: May be replace name of the constant to something meaningful

File: SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                    }
+                })
+            }
+        }
+    }
+}
+
+struct SwiftDataEmployeesView: View {
+    @State var department: Department
+    @State var vm: SwiftDataViewModel
```
Comment (bguchinov) [inline SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift:50]: Islam how do you think is this okay to name variables like that? I think if this the same as id for example, so its okay

File: SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    @State private var employeeName: String = ""
+    
+    var body: some View {
+        HStack {
+            TextField("id", text: $id)
+                .textFieldStyle(.roundedBorder)
+            TextField("name", text: $employeeName)
+                .textFieldStyle(.roundedBorder)
+            Button(action: {
+                let emp = Employee(id: Int(id) ?? 1, name: employeeName)
```
Comment (bguchinov) [inline SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift:61]: rename const

**Review by bguchinov — CHANGES_REQUESTED** [review]

File: SwiftUIStudy/Core/Managers/SwiftDataManager.swift (hunk 1 above)
Comment (MaksimK193) [reply SwiftUIStudy/Core/Managers/SwiftDataManager.swift:52]: fixed

File: SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift (hunk 2 above)
Comment (MaksimK193) [reply SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift:23]: fixed

File: SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift (hunk 4 above)
Comment (MaksimK193) [reply SwiftUIStudy/Screens/SwiftData/SwiftDataView.swift:61]: fixed

**Pull request merged**